src/tpm/kmyth_seal_unseal_impl.c

```diff
@@ -28,21 +27,90 @@
  */
 extern const cipher_t cipher_list[];
 
+//############################################################################
+// convert_string_to_digest()
+//############################################################################
+static int convert_string_to_digest(char *str, TPM2B_DIGEST * digest)
+{
+  // substring to holds 2 hex values at a time
+  char substr[3];
+
+  substr[2] = '\0';
+
+  size_t strlength = strlen(str);
+
+  if (strlength != (size_t) 132)
+  {
+    return 1;
+  }
+  // initializes buffer with all 0 hex values
+  unsigned char expectedPolicyBuffer[132] = { 0x00 };
+  unsigned long ul;
+
+  // iterates through each pair of hex values and fills the
+  //  buffer with values indicated in the string
+  for (size_t i = 0; i < strlength; i += 2)
+  {
+    strncpy(substr, &str[i], 2);
+    ul = strtoul(substr, NULL, 16);
+    expectedPolicyBuffer[i / 2] = ul;
+  }
+
+  // converts the byte array into a TPM2B_DIGEST struct
+  memcpy(digest, expectedPolicyBuffer, sizeof(expectedPolicyBuffer));
+  return 0;
+}
+
+//############################################################################
```

Comment:
As the previous function, this on seems like more of a "formatting" or "TPM interface" utility. My preference would be to move it, say, to utils/src/formatting_tools.c.